app/services/spam_report_service.rb

```diff
@@ -1,14 +1,29 @@
 class SpamReportService < ApplicationService
-  attr_reader :delivery_attempt
+  attr_reader :delivery_attempt_id, :email_address
 
-  def initialize(delivery_attempt)
-    @delivery_attempt = delivery_attempt
+  def initialize(delivery_attempt_id, email_address)
+    @delivery_attempt_id = delivery_attempt_id
+    @email_address = email_address
   end
 
   def call
-    subscriber_id = delivery_attempt.email.subscriber_id
-    subscriber = Subscriber.find(subscriber_id)
+    return if email.marked_as_spam?
+
+    UnsubscribeAllService.call(subscriber, :marked_as_spam)
+    Metrics.marked_as_spam
+    email.update!(marked_as_spam: true)
+  rescue ActiveRecord::RecordNotFound
     UnsubscribeAllService.call(subscriber, :marked_as_spam)
-    delivery_attempt.email.update!(marked_as_spam: true)
+    Metrics.marked_as_spam
```

Comment:
It's great that we're handling the 404s but I think this metric becomes somewhat confused if we'll populate it for all delivery attempts that 404 but only for emails we've not already marked as spam.